### ISSUE TITLE
WebSocketClient Issue

### ISSUE THREAD

**Issue by NamedDaniel** [DV8FromTheWorld/JDA#1322]
Hi.
I hope you can help me:
I permanently get the same error message:

> [JDA [0 / 1] Gateway-Worker 1] WARN WebSocketClient - Missed 2 heartbeats! Trying to reconnect...

As well, I tried to make an 'exit'  command for the Console.
But every time I try to execute it, I get this:

```
java.io.IOException: Stream closed
	at java.io.BufferedReader.ensureOpen(Unknown Source)
	at java.io.BufferedReader.readLine(Unknown Source)
	at java.io.BufferedReader.readLine(Unknown Source)
	at de.nameddaniel.bot.main.Main.lambda$0(Main.java:62)
	at java.lang.Thread.run(Unknown Source)
[JDA [0 / 1] Gateway-Worker 1] ERROR WebSocketClient - Was unable to schedule next packet due to rejected execution by threadpool
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@722cda80 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@5fec53b2[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 8153]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.reject(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(Unknown Source)
	at net.dv8tion.jda.internal.requests.WebSocketSendingThread.scheduleSentMessage(WebSocketSendingThread.java:89)
	at net.dv8tion.jda.internal.requests.WebSocketSendingThread.scheduleNext(WebSocketSendingThread.java:169)
	at net.dv8tion.jda.internal.requests.WebSocketSendingThread.run(WebSocketSendingThread.java:157)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

Here's my shutdown method. Main,java:62 is highlighted.

```java
public void shutdown() {
		
         new Thread(() -> {
			
			String line = "";		
			BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));			
			
			try {
				while((line = reader.readLine()) != null) { //Line 62
					if(line.equalsIgnoreCase("exit")) {
						if(shard != null) {
							shard.setStatus(OnlineStatus.OFFLINE);
							shard.shutdown();
							System.out.println("Bot wurde heruntergefahren");
						}
						reader.close();
					}
					else {
						System.out.println("'Exit' zum herunterfahren");
					}
				}
			}
			catch(IOException e) {
				System.out.println("Main");
				e.printStackTrace();
			}
			
		}).start();
```

I hope, you can help me :(

**Comment by MinnDevelopment** [comment]
It looks like you're using some custom thread pool configuration. It would be extremely helpful for us if you would actually fill out the issue template instead of just removing it. The IOException you included doesn't seem like it's related to JDA in any way, but I can answer why it happens:

```java
Reader reader = myReader();
reader.close();
reader.nextLine() // can't read if you have a closed stream
```
 This is what you are doing in your loop here.

**Comment by Andre601** [comment]
Also, in addition should you next time follow one of the issue templates in order to provide as much information as needed.

**Comment by MinnDevelopment** [comment]
The `RejectedExecutionException` here doesn't seem to be an issue, it means that JDA was probably shutdown and some tasks were still queued up which are now rejected and won't be sent. Regardless, this shouldn't be a problem since the messages can't be sent anyway if the connection was shutdown.

That's really all I can say since you didn't provide enough information for me to diagnose the situation. You provided an incomplete code sample with little detail and directly ignored our issue template.

The warning that you mentioned about the missing heartbeats could mean anything really. Missing heartbeats can be caused by a number of different scenarios such as:

- An outdated JDA version
- Blocking the event thread
- Connection issues
- Discord issues
- WebSocket issues
- Threading issues

I just recently fixed an issue related to this here #1294 and since you didn't tell us your version, I'll assume it's probably that.

Feel free to open a new issue, this time filling out the template with the necessary details.

Duplicate of #1294
